pkgs/tools/security/nwaku/default.nix

```
@@ -0,0 +1,97 @@
+{
+  stdenv, lib, fetchurl, fetchFromGitHub,
+  nim, cmake, which, writeScriptBin,
+  # Options: wakunode1, wakunode2, wakubridge
+  makeTargets ? [ "wakunode2" ],
+  # WARNING: CPU optmizations that make binary not portable.
+  nativeBuild ? false
+}:
+
+assert nim.version == "1.6.12";
+
+stdenv.mkDerivation rec {
+  pname = "nwaku";
+  version = "0.16.0";
+  commit = "03f0de36";
+  name = "${pname}-${version}-${commit}";
+
+  src = fetchFromGitHub {
+    owner = "waku-org";
+    repo = "nwaku";
+    rev = "v${version}";
+    sha256 = "sha256-yj+wNqo25BrLm+RjHelbnhg/Cq+m8Sayk+ycDw+Tksk=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ nim ]
+    ++ lib.optionals stdenv.isDarwin [ cmake ];
+
+  buildInputs = let
+    # Fix for Nim compiler calling 'git rev-parse' and 'lsb_release'.
+    fakeGit = writeScriptBin "git" "echo $commit";
```

```suggestion
    fakeGit = writeScriptBin "git" "echo v${version}";
```
we can cheat that easily

---

But I don't want the version. I want the commit. Why would I change it like that?

---

it would be easier to update and maintain and only adds very minimal information

---

I'm going to be maintaining this and I'll somehow cope with the immense burden of updating the `commit` attribute.